doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+    
+    ![PIM_SPT](pim_images/PIM_SM_Shortest_Path_Tree.png "Figure 2: IP Multicast deployment with PIM-ASM Shortest Path Tree")
+ 
+  __Figure 2: IP Multicast Deployment with PIM-ASM Shortest Path Tree__
+
+# 3 Design 
+
+## 3.1 Overview
+
+The following diagram depicts the interactions among modules in SONIC to achieve PIM ASM functionality 
+
+
+  ![PIM_ASM_DESIGN_OVERVIEW](pim_images/PIM_ASM_DESIGN_OVERVIEW.png)
+
+   
+   __Figure 3: PIM ASM Architecture in SONiC__
+
+The interactions between FRR PIM, Linux kernel, ipmcFpmSyncd and ipmcOrch for updating a multicast (*,G) and (S,G) route entry from PIM to the hardware are explained below:
+
+A small change has been introduced in ipmFpmsyncd for handling the KAT timer expiry of (S,G) routes which is explained in case 3 below.
+
+#### Case 1: Flow for (*,G) route update
+
+1. PIM ASM creates a (*,G) entry in 2 scenarios. First, when LHR node receives an IGMP report message and second, when RP and other transit routers receives a PIM join from the downstream. PIMD listens on a socket for receiving the PIM and IGMP control packets.
+   
+   1.1.(a) When the LHR receives an IGMP Report, it is passed to the Linux kernel.  Kernel delivers the IGMP Report payload to the Pimd process.  Pimd creates a IGMP group membership entry and checks if the RP address is reachable.  If reachable, Pimd creates a (\*,G) entry with the RPF interface as its IIF and updates the (\*,G) entry's OIF with the interface on which the IGMP Report is received.
+   Data packets are needed by Pimd to do SPT switchover in case of LHR, To achieve this, the OIF list is updated with the pimreg interface. <br> <br>
+
+    1.1.(b) When the RP or an intermediate PIM router receives a PIM (\*,G) Join message, the message is passed to the Linux kernel and kernel delivers the PIM Join message and its associated payload to Pimd process.  PIM queries Zebra for RPF validation and then creates a (\*,G) entry with the RPF interface as its IIF. In case of RP, the IIF interface is the interface on which the RP address is configured. PIM queries IGMP database (inside Pimd) and updates the (\*,G) entry's OIF list. To complete the registration process RP needs the data packet in Pimd, so to get the multicast data in Pimd the pimreg interface is updated in the OIF list.  <br> <br>
+    
+2. After PIM creates the (\*,G) entry with the required information, PIM updates Linux kernel using a socket and kernel creates a mroute entry. <br> <br>
```

Comment:
at RP (*,G) entry will not have IIF , PIM protocol will have entry with IIF NULL, is that entry gets updated/sent to kernel? or only SG will be updated to kernel, please add more detail on this case at RP